mountpoint-s3-fs/examples/json_arguments.rs

```diff
@@ -0,0 +1,264 @@
+                        tracing::warn!("Failed to detect network throughput. Using {DEFAULT_THROUGHPUT} gbps: {e:?}");
+                        Ok(DEFAULT_THROUGHPUT)
+                    }
+                }
+            }
+            ThroughputConfig::IMDSLookUp { ec2_instance_type } => {
+                autoconfigure::get_maximum_network_throughput(ec2_instance_type).context("Unrecognized instance ID")
+            }
+        }
+    }
+}
+
+fn load_config<P: AsRef<Path>>(path: P) -> Result<ConfigOptions> {
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+    let config: ConfigOptions = serde_json::from_reader(reader)?;
+    if config.channels.len() != 1 {
+        return Err(anyhow!("Exactly one channel must be configured"));
+    }
+    Ok(config)
+}
+
+// Simplified manifest processing function
+fn process_manifests(config: &ConfigOptions) -> Result<Option<PathBuf>> {
+    // Skip if no manifests or no channels
+    if config.channels.is_empty() || config.channels[0].manifest_path.is_none() {
+        return Ok(None);
+    }
+
+    let channel = &config.channels[0];
+
+    // Skip if no CSV path provided
+    let csv_path = match &channel.manifest_path {
+        Some(path) => path,
+        None => return Ok(None),
+    };
+    // Generate manifest path and check if it exists
+    let db_path = Path::new(&config.metadata_store_dir)
+        .join("manifests")
+        .join(format!("{}.db", channel.directory_name));
+    fs::create_dir_all(db_path.parent().unwrap())?;
+    println!("Creating manifest for channel {}", channel.directory_name);
+    let start = Instant::now();
+    ingest_manifest(csv_path, &db_path)?;
+    println!("Created manifest in {:?}", start.elapsed());
+    Ok(Some(db_path.clone()))
+}
+
+fn setup_logging(config: &ConfigOptions) -> Result<()> {
+    let _logging = init_logging(config.build_logging_config())?;
+    let _metrics = metrics::install();
+    Ok(())
+}
+
+fn mount_filesystem(config: &ConfigOptions, manifest_path: Option<PathBuf>) -> Result<()> {
+    // Create the Mountpoint configuration
+    let mp_config = MountpointConfig::new(
+        config.build_fuse_session_config()?,
+        config.build_filesystem_config(manifest_path)?,
+        config.build_data_cache_config(),
+    );
+
+    // Create the client and runtime
+    let client = config
+        .build_client_config()?
+        .create_client(Some(&config.build_s3_path()?))
+        .context("Failed to create S3 client")?;
+    let runtime = Runtime::new(client.event_loop_group());
+
+    // Create and run the FUSE session
+    let fuse_session = mp_config
+        .create_fuse_session(config.build_s3_path()?, client, runtime)
+        .context("Failed to create FUSE session")?;
+
+    // Join the session and wait until it completes
+    fuse_session.join().context("Failed to join session")
+}
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about = "Mountpoint configuration validator")]
```

Comment:
Consider renaming the example to something like `mount_from_config` and using:
```suggestion
#[clap(version, about = "Mountpoint launcher from configuration file")]
```